src/main/java/se/sundsvall/emailsender/service/EmailService.java

```diff
@@ -133,4 +134,30 @@ Optional<String> decodeBase64(final String s) {
 			return Optional.empty();
 		}
 	}
+
+	/**
+	 * RFC5322 states that Message-ID's should be separated by CRLF,
+	 * this method formats a list of values using \r\n to fulfil the requirements.
+	 *
+	 * @param values The list of strings
+	 * @return The formatted string
+	 */
+	String formatHeader(final List<String> values) {
+		return values.stream()
+			.reduce((a, b) -> a + "\r\n" + b)
+			.orElse("");
+	}
+
+	void applyCustomHeaders(final MimeMessage message, final SendEmailRequest request) throws MessagingException {
+		if (request.getHeaders() == null) {
```

Comment:
Use: `Optional.ofNullable(request.getHeaders()).ifPresent(...) ` instead?

Pros: One branch less to cover in your tests.